src/dc_receive_imf.rs

```diff
@@ -1706,13 +1706,14 @@ async fn apply_group_changes(
             send_event_chat_modified = true;
         }
     } else if let Some(contact_id) = removed_id {
-        if chat_id
-            .update_timestamp(context, Param::MemberListTimestamp, sent_timestamp)
-            .await?
-        {
-            chat::remove_from_chat_contacts_table(context, chat_id, contact_id).await?;
-            send_event_chat_modified = true;
-        }
+        // in contrast to `Chat-Group-Member-Added` we do not reconstruct the whole state from To: on removing -
+        // otherwise out-of-order removals won't work as members are re-added quickly.
```

Comment:
re-added?

Reply:
Adding members problem is the same, and it can be solved the same way: we can just add a single member mentioned in the header instead of rebuliding the whole list, and it will work properly with reordering.

Long term solution is to reconstruct the whole list from `To:` both on member addition and on removal (for removal the deleted user should be moved from `To:` to `Bcc:`).

There is no reason why reordering when adding members is solved using timestamps and reordering when removing members is solved by actually looking at the `Chat-Member-*` header contents, it could have been the other way.

Reply:
> Adding members problem is the same, and it can be solved the same way: we can just add a single member mentioned in the header instead of rebuliding the whole list

iirc, we did that in the past and switched to the current implementation for $reasons.

but yes, that all should be revisited at some point :)